Userland/Applications/PackageManager/InstalledPort.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Function.h>
+#include <Applications/PackageManager/InstalledPort.h>
+#include <LibCore/File.h>
+#include <LibCore/System.h>
+
+ErrorOr<HashMap<String, InstalledPort>> InstalledPort::read_ports_database()
+{
+    auto file = TRY(Core::File::open("/usr/Ports/installed.db"sv, Core::File::OpenMode::Read));
+    auto buffered_file = TRY(Core::InputBufferedFile::create(move(file)));
+    auto buffer = TRY(ByteBuffer::create_uninitialized(PAGE_SIZE));
+
+    HashMap<String, InstalledPort> ports;
+    while (TRY(buffered_file->can_read_line())) {
+        auto line = TRY(buffered_file->read_line(buffer));
+        if (line.starts_with("dependency"sv)) {
+            auto parts = line.split_view(' ');
+            auto type = InstalledPort::Type::Dependency;
+            // FIXME: Add versioning when printing these ports!
+            auto name = TRY(String::from_utf8(parts[1]));
```

Comment:
This is not correct, the syntax is

`dependency [port] [dependency]`

Which states that `[dependency]` was installed on behalf of `[port]`.